src/functions/middleware/user.ts

```diff
@@ -0,0 +1,7 @@
+import { defineMiddleware } from 'astro/middleware'
+
+export const userMiddleware = defineMiddleware(async ({ locals }, next) => {
+  locals.user = await locals.backend.auth.getAuthenticatedUser.query()
```

Comment:
Not sure if we should do lazy evaluation on this or not. i.e.

```ts
  locals.getUser = memoize(() => locals.backend.auth.getAuthenticatedUser.query())
```

this is so that we don't need to invoke backend when an endpoint doesn’t need authentication

Reply:
if do this way, we need unique key to cache different user as well. prevent Steam cache incident

Reply:
@rayriffy i believe `locals` are not shared between requests. [per docs](https://docs.astro.build/en/guides/middleware/#locals:~:text=locals%20is%20an%20object%20that%20lives%20and%20dies%20within%20a%20single%20Astro%20route%3B%20when%20your%20route%20page%20is%20rendered%2C%20locals%20won%E2%80%99t%20exist%20anymore%20and%20a%20new%20one%20will%20be%20created.%20Information%20that%20needs%20to%20persist%20across%20multiple%20page%20requests%20must%20be%20stored%20elsewhere.):

> `locals` is an object that lives and dies within a single Astro route; when your route page is rendered, locals won’t exist anymore and a new one will be created. Information that needs to persist across multiple page requests must be stored elsewhere.

Reply:
i used `mem` instead of `lodash/memorize`. seems lodash's function cannot mem async functions